src/runtime/plan/suggestion.ts

```diff
@@ -102,12 +102,39 @@ export class Suggestion {
     return (this.hash === other.hash) && (this.descriptionText === other.descriptionText);
   }
 
+  isEqual(other: Suggestion): boolean {
+    return this.isEquivalent(other) &&
+          this.rank === other.rank &&
+          this._isSameSearch(other) &&
+          this._isSameDescription(other) &&
+          this._isSameVersions(other);
+  }
+
+  _isSameSearch(other: Suggestion): boolean {
+    return this.searchGroups.length === other.searchGroups.length &&
```

Comment:
Should you consider using a Set for searchGroups ?  This might better encapsulate the semantics and simplify this ccode.

Reply:
Did you mean Set<string[]> or Set<string>[] (or Set<Set<string>>)? :)
I tried either and actually it didn't seem to make the code simpler...

I totally agree that the current implementation is suboptimal, but i'm thinking maybe it should be more like the `Search` class (recipe/search.ts), similar to the recipe, not sure yet. 
Anyway, it is outside the scope of this PR, so I prefer to leave it unchanged for now, and give it some extra thought later, if that's ok with you.